site/content/docs/private-locations/checkly-agent-guide.md

```diff
@@ -0,0 +1,164 @@
+
+* All of the agents in the private location have failed
+* There is insufficient agent capacity in the private location
+
+Checks will stay queued for 6 minutes waiting for an available agent. If there are no agents available or the available agents are so busy that they cannot process all of the pending checks, the pending checks older than 6 minutes will be lost.
+
+Agents can be scaled up and scaled out. Scaling up means adding additional CPU and memory allocations to existing agent containers. Scaling out means adding additional agent containers.
+
+### Scale up
+
+Scaling up of individual agents is generally memory-constrained. For reference, a browser check requires about 1.5GB of RAM and an API check requires about 150MB. The number of concurrent checks allowed to run on a single agent is controlled by the `JOB_CONCURRENCY` environment variable which defaults to 1 and can be increased to 10. We do not automatically separate browser and API checks on an agent. If you are running only browser checks or a combination of browser and API checks in a private location, the formula (rounded down) is:
+
+`JOB_CONCURRENCY = Container memory allocation (GB) / 1.5`
+
+For example: if your container has 8GB of memory allocated, you should set `JOB_CONCURRENCY` to 5 (8GB / 1.5GB per check).
+
+If your private location is only running API checks:
+
+`JOB_CONCURRENCY = Container memory allocation (GB) / 0.15`
+
+For example: if your container has 1GB of memory allocated, you should set `JOB_CONCURRENCY` to 6 (1GB / 0.15GB per API check). With 1.5GB of memory you can set `JOB_CONCURRENCY` to the maximum value of 10.
+
+ ### Scale out
+
+To determine the number of agents you need in a private location you first need to know the number of checks assigned to the location and their frequency. API checks can be scheduled as frequently as every 10 seconds and browser checks as frequently as every 1 minute. Based on your configuration, you should be able to estimate how many checks will run per minute in your private location. Checks have a maximum running time of 30 seconds. This means that in the worst case scenario, a checkly agent with a `JOB_CONCURRENCY` of 1 can run two checks per minute. In an average configuration this will be higher and API checks are generally faster than browser checks.
+
+Once you have an idea of how many checks will be running and you know the per-agent `JOB_CONCURRENCY` limit based on the agent container memory allocation, you can estimate the number of agents required as `agents = (JOB_CONCURRENCY / Checks per minute / 2) + 1`. This should give you a safe amount of overhead to run your checks and the `+ 1` is to ensure redundant capactity in case of rolling upgrades or an agent failure. 
+
+## Installing and configuring the Checkly agent:
```

Comment:
Maybe it makes sense to put this before the "Redundancy and scaling" section?

I think users will need to get the Checkly Agent running before they worry about scaling and redundancy.